packages/phoenix-ng/projects/phoenix-app/cypress/e2e/cms.test.ts

```diff
@@ -0,0 +1,171 @@
+describe('CMS', () => {
+  it('should match the whole page', () => {
+    cy.visit('/cms').then(() => {
+      cy.contains('This may take a while').should('not.be.visible');
+      // in order to make the detector appear which loads in 2 - 2.5 seconds
+      cy.wait(2500);
+      cy.document().toMatchImageSnapshot();
+    });
+  });
+
+  describe('Phoenix Menu', () => {
+    it('should have options to Save and Load state', () => {
+      cy.get('.phoenix-menu-item-right .item-settings').click();
+
+      cy.contains('Save state').should('exist');
+      cy.contains('Load state').should('exist');
+
+      cy.get('.phoenix-menu-item-right .item-config-backdrop').click();
+    });
+
+    it('should display correct options under event data dropdown', () => {
+      cy.get('.phoenix-menu-item-right > button').eq(0).click();
+      cy.get('.phoenix-menu-item-right > button').eq(1).click();
```

Comment:
Unfortunately, the order of these buttons could change depending on if the geometry or event data loads first.

Maybe you can use a selector like this:

```ts
cy.get("app-phoenix-menu").contains("Event Data").closest(".item-expand");
```

Reply:
If you can, the best approach would be to add `data-testid`s to Phoenix menu items. But that will need some changing in the Phoenix menu code which could be complex.

Reply:
You could also create a function:

```ts
const getPhoenixMenuItem = (itemName: string) => {
  return cy.get("app-phoenix-menu").contains(itemName);
}

getPhoenixMenuItem("Event Data").closest(".item-expand").click();
```

Reply:
The above comments are for all selections in Phoenix menu.